lib/pure/strutils.nim

```diff
@@ -2860,6 +2860,79 @@ proc isNilOrWhitespace*(s: string): bool {.noSideEffect, procvar, rtl,
   ## Alias for isEmptyOrWhitespace
   result = isEmptyOrWhitespace(s)
 
+
+since (1, 1):
+  template toLowerAsciiInPlace(c: var char) =
+    if c in {'A'..'Z'}: c = chr(ord(c) + (ord('a') - ord('A')))
+
+  template toUpperAsciiInPlace(c: var char) =
+    if c in {'a'..'z'}: c = chr(ord(c) - (ord('a') - ord('A')))
+
+
+  func toLowerAsciiInPlace*(s: var string) {.inline.} =
+    ## Converts string `s` into lower case.
+    ##
+    ## This works only for the letters ``A-Z``. See `unicode.toLower
+    ## <unicode.html#toLower,string>`_ for a version that works for any Unicode
+    ## character.
+    ##
+    ## See also:
+    ## * `normalize proc<#normalize,string>`_
+    runnableExamples:
+      var stringy = "ABCDEF"
+      toLowerAsciiInPlace(stringy)
+      doAssert stringy == "abcdef"
+      var strng = "NIM"
+      toLowerAsciiInPlace(strng)
+      doAssert strng == "nim"
+    var i = 0
+    for c in mitems(s):
+      toLowerAsciiInPlace(c)
+      s[i] = c
+      inc i
```

Comment:
```suggestion
      c = toLowerAscii(c)
```

You're using `mitems`, which directly modifies the elements of the passed string.